### DISCUSSION TITLE
GUI dependency

### DISCUSSION THREAD

**Discussion by costaht** [ImageMagick/ImageMagick#5510] [Help] [answered]
Hi guys. I wrote a script on my laptop that uses convert and composite, but now I'm trying to run it on a computer that has no GUI installed. When installing ImageMagick it tries to install Gnome and everything else because it relies on a Gui.
Is there a cli only version of ImageMagick?
Thank you

**Comment by fmw42** [comment]
What did you install?  Imagemagick is basically command line. There is a small GUI on Unix with minimal functionality. For installs, see https://imagemagick.org/script/download.php or https://imagemagick.org/script/install-source.php#unix

**Comment by snibgo** [comment]
Perhaps the OP used a package manager that recognises that one component of IM, `display` needs a GUI interface to the operating system. But `display` is only a minor component, and IM is useful without it.

So this may be a problem with a package manager.

**Reply by costaht** [reply]
That's exactly what I'm doing. I'm was installing it using dnf. Now I've installed IM rpm manually and I'm trying to install all its dependencies without adding the graphical components

**Comment by costaht** [comment, marked as answer]
Here's the solution I came up with. I downloaded the Appimage from GitHub releases and saved it in my /usr/bin as `magic`, set the permission to executable, and before each command (composite, convert) I added the word `magic`.
